x/gov/types/msgs.go

```diff
@@ -67,11 +50,18 @@ func (msg MsgSubmitProposal) ValidateBasic() sdk.Error {
 	if msg.InitialDeposit.IsAnyNegative() {
 		return sdk.ErrInvalidCoins(msg.InitialDeposit.String())
 	}
-	return nil
+	if !IsValidProposalType(msg.Content.ProposalType()) {
```

Comment:
We're not modifying the function @mossid. In fact, I _intentionally_ did this because before we had to check the type everywhere (very error prone) and now we only check it in one place. Any proposal type now only has to call `RegisterProposalType` and that's it!

Reply:
Oh sorry, I thought `IsValidProposalType` was not changed from the original code, I found it. Looks great!